usd/src/usd_parser/utils.cc

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "sdf/usd/usd_parser/utils.hh"
+
+#include <pxr/usd/usdShade/input.h>
+#include <pxr/usd/usdShade/material.h>
+#include <pxr/usd/usdShade/shader.h>
+
+#include <ignition/common/Filesystem.hh>
+#include <ignition/common/Util.hh>
+
+#include "sdf/Pbr.hh"
+
+namespace sdf
+{
+  // Inline bracke to help doxygen filtering.
+  inline namespace SDF_VERSION_NAMESPACE {
+  //
+  namespace usd
+  {
+  /////////////////////////////////////////////////
+  std::string removeSubStr(const std::string &_str, const std::string &_substr)
+  {
+    std::string result = _str;
+    size_t pos = std::string::npos;
+    if ((pos = result.find(_substr) )!= std::string::npos)
+    {
+      result.erase(pos, _substr.length());
+    }
+    return result;
```

Comment:
The docs says it should remove all substring appearences, but I think this code only removes the first.